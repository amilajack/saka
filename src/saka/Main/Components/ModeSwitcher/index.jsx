import { h } from 'preact';
import { suggestions } from 'src/suggestion_engine/server/providers/mode.js';
import 'scss/styles.scss';

const Icon = ({ icon, color }) => (
  <i className="material-icons" aria-hidden="true" style={{ color }}>
    {icon}
  </i>
);

export default ({ mode, setMode }) => {
  const validModes = suggestions.map(suggestion => {
    return (
      <div
        className="mode-switcher-icon"
        style={
          suggestion.mode === mode
            ? `background-color: white; border-top: 3px solid  ${
                suggestion.fadedColor
              };`
            : {}
        }
        onClick={() => setMode(suggestion.mode)}
      >
        <Icon icon={suggestion.icon} color={suggestion.fadedColor} />
      </div>
    );
  });

  return <div className="mode-switcher-wrapper">{validModes}</div>;
};
