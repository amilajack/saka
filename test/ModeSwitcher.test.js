import ModeSwitcher from '@/saka/Main/Components/ModeSwitcher/index.jsx';
import { render } from 'preact-testing-library';
import { h } from 'preact';

describe('ModeSwitcher component ', () => {
  it('should render tabs with selected tab colored, rest of tabs gray', () => {
    const props = {
      mode: 'tab',
      setMode: () => {}
    };

    const { getByText } = render(<ModeSwitcher {...props} />);

    getByText('tab');
    getByText('restore_page');
    getByText('bookmark_border');
    getByText('history');
    getByText('timelapse');
  });
});
