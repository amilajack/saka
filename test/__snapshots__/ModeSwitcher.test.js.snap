// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModeSwitcher component  should render tabs with selected tab colored, rest of tabs gray 1`] = `
<i
  aria-hidden="true"
  class="material-icons"
  id="icon"
  style="color: rgba(55, 126, 184, 0.44);"
>
  tab
</i>
`;

exports[`ModeSwitcher component  should render tabs with selected tab colored, rest of tabs gray 2`] = `
<i
  aria-hidden="true"
  class="material-icons"
  id="icon"
  style="color: rgba(153, 153, 153, 0.44);"
>
  restore_page
</i>
`;

exports[`ModeSwitcher component  should render tabs with selected tab colored, rest of tabs gray 3`] = `
<i
  aria-hidden="true"
  class="material-icons"
  id="icon"
  style="color: rgba(153, 153, 153, 0.44);"
>
  bookmark_border
</i>
`;

exports[`ModeSwitcher component  should render tabs with selected tab colored, rest of tabs gray 4`] = `
<i
  aria-hidden="true"
  class="material-icons"
  id="icon"
  style="color: rgba(153, 153, 153, 0.44);"
>
  history
</i>
`;

exports[`ModeSwitcher component  should render tabs with selected tab colored, rest of tabs gray 5`] = `
<i
  aria-hidden="true"
  class="material-icons"
  id="icon"
  style="color: rgba(153, 153, 153, 0.44);"
>
  timelapse
</i>
`;
